.github/workflows/ci-pullrequest-main.yml

```diff
@@ -0,0 +1,73 @@
+name: Mock Classifier CI Pipeline
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  BuildTest:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.0.x
+        
+    - name: Dotnet Restore Tools
+      run: dotnet tool restore --tool-manifest "src/.config/dotnet-tools.json"
+        
+    - name: Dotnet Restore
+      run: dotnet restore src/*.sln
+        
+    - name: Dotnet Format
```

Comment:
I think this might be more successful as a pre-commit hook...

Reply:
dotnet format --verify-no-changes might be the one to fail the build if the formatting is wrong.